### PR TITLE
feat: add folder delete, move modal

### PR DESCRIPTION
**Pull request by yuda110**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci`, minor refactoring, etc.)
- [ ] Need discussion
- [ ] Not that difficult
- [ ] Approved feature branch merge to master

### Description
tree
스크린샷 2024-09-19 오후 2 38 42

modals
스크린샷 2024-09-19 오후 2 39 47
스크린샷 2024-09-19 오후 2 39 52

### CODE REVIEW

File: apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderDeleteModal.vue [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+import { useDashboardMainPageStore } from '@/services/dashboards/stores/dashboard-main-page-store';
+import type { DashboardTreeDataType } from '@/services/dashboards/types/dashboard-folder-type';
+
+
+
+const DELETE_TABLE_FIELDS = [
+    { name: 'name', label: 'Name' },
+    { name: 'location', label: 'Location' },
+];
+type FolderDeleteParams = PublicFolderDeleteParameters | PrivateFolderDeleteParameters;
```
Comment (WANZARGEN) [inline apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderDeleteModal.vue:33]: nitpick: What about to add file under schema/dashboard which includes those kind of reusable union types?

**Review by sulmoJ — APPROVED** [review]
LGTM!

File: apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderDeleteModal.vue (hunk 1 above)
Comment (yuda110) [reply apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderDeleteModal.vue:33]: okay!!

File: apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderDeleteModal.vue [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+                type: 'DASHBOARD',
+            });
+        }
+    });
+    return _tableItems;
+};
+const getSelectedFolderItems = (selectedFolderId: string): DataTableItem[] => {
+    const _tableItems: DataTableItem[] = [];
+    const _selectedFolder = state.dashboardTreeData.find((d) => d.id === selectedFolderId);
+    if (!_selectedFolder) return _tableItems;
```
Comment (WANZARGEN) [inline apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderDeleteModal.vue:97]: nitpick: I think this flow is pretty embarrassing - User selects folder, but if it's absent on tree data, then just show empty table.🥲

File: apps/web/src/services/dashboards/stores/dashboard-main-page-store.ts [hunk 3]
```diff
@@ -1,17 +1,138 @@
...
+    const _selectedIdList = Object.keys(selectedIdMap).filter((key) => selectedIdMap[key]);
+    const _selectedNodeList: TreeNode<DashboardTreeDataType>[] = [];
+    _selectedIdList.forEach((id) => {
+        const _node = dashboardTreeData.find((node) => node.data.id === id);
+        if (!_node) return;
+        if (_node?.data.type === 'FOLDER') {
+            // get folder id only if all children are selected
+            if (_node.children?.every((child) => selectedIdMap[child.id])) {
+                _selectedNodeList.push(_node);
+            } else {
```
Comment (WANZARGEN) [inline apps/web/src/services/dashboards/stores/dashboard-main-page-store.ts:87]: question: Oh, if children are all selected, then automatically remove folder? awesome!

File: apps/web/src/services/dashboards/stores/dashboard-main-page-store.ts [hunk 4]
```diff
@@ -1,17 +1,138 @@
...
+                const _childrenDashboards = _node.children?.filter((child) => selectedIdMap[child.id]);
```
Comment (WANZARGEN) [inline apps/web/src/services/dashboards/stores/dashboard-main-page-store.ts:88]: suggestion: I think it's pretty inefficient..🥲 
what about to add flag like `removeFolder`, and do not use `Array.prototype.every()` method in the if statement? 
it loops twice for else cases, and I think most cases will be the else case.

**Review by WANZARGEN — APPROVED** [review]
Great job on this! I've left a few small suggestions that might help improve the code. Let me know what you think!

**Pull request merged**